python/cpp/encoder.cc

```diff
@@ -13,20 +13,25 @@ namespace ctranslate2 {
 
       EncoderForwardOutput
       forward_batch(const std::variant<BatchTokens, BatchIds, StorageView>& inputs,
-                    const std::optional<StorageView>& lengths) {
+                    const std::optional<StorageView>& lengths,
+                    const std::optional<BatchIds>& token_type_ids
+                    ) {
         std::future<EncoderForwardOutput> future;
 
         switch (inputs.index()) {
         case 0:
-          future = _pool->forward_batch_async(std::get<BatchTokens>(inputs));
+          future = (token_type_ids) ? _pool->forward_batch_async(std::get<BatchTokens>(inputs), token_type_ids.value()) 
+                                    : _pool->forward_batch_async(std::get<BatchTokens>(inputs));
```

Comment:
Is it possible to avoid duplicating the method call? Something like this:

```cpp
future = _pool->forward_batch_async(std::get<BatchTokens>(inputs), token_type_ids.value_or(std::vector<std::vector<size_t>>()));
```